policy/loader.go

```diff
@@ -6,58 +6,59 @@ import (
 
 	"github.com/open-policy-agent/conftest/downloader"
 
-	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/loader"
 )
 
 // Loader handles the retrieval of all rego policies and related data.
 type Loader struct {
 	PolicyPaths []string
 	DataPaths   []string
 	URLs        []string
-
-	test bool
+	Tracing     bool
 }
 
-// SetTestLoad configures the loader to load Rego test files as well
-func (l *Loader) SetTestLoad(test bool) *Loader {
-	l.test = test
-	return l
-}
+// Load returns an Engine after loading all of the specified policies and data paths.
+// If URLs are specified, Load will first download all of the policies at the spcified URLs.
+func (l *Loader) Load(ctx context.Context) (*Engine, error) {
 
-// Load retrieves policies from several locations:
-// first it checks for any remote sources of policies and downloads
-// the policies into the given policy paths.
-// After retrieving the policies from the remote sources, all .rego, .json and .yaml
-// files are recursively retrieved from disk and loaded into
-// a rego Compiler and Store respectively.
-func (l *Loader) Load(ctx context.Context) ([]string, storage.Store, error) {
 	// Downloaded policies are put into the first policy directory specified
 	for _, url := range l.URLs {
 		sourcedURL, err := downloader.Detect(url, l.PolicyPaths[0])
 		if err != nil {
-			return nil, nil, fmt.Errorf("detect policies: %w", err)
+			return nil, fmt.Errorf("detect policies: %w", err)
 		}
 
 		if err := downloader.Download(ctx, l.PolicyPaths[0], []string{sourcedURL}); err != nil {
-			return nil, nil, fmt.Errorf("update policies: %w", err)
+			return nil, fmt.Errorf("update policies: %w", err)
 		}
 	}
 
-	var regoFiles []string
-	var err error
-	if l.test {
-		regoFiles, err = ReadFilesWithTests(l.PolicyPaths...)
-	} else {
-		regoFiles, err = ReadFiles(l.PolicyPaths...)
+	paths := append(l.PolicyPaths, l.DataPaths...)
+	result, err := loader.All(paths)
```

Comment:
Does this deal with the differences in test vs non-test rego files? I guess the Query defines what gets executed and it is not the worst to just load all files.

Reply:
Out of the box, it does not. If we felt strongly enough we could add a filter, but to your last point it didn't seem worth it to filter them out.